### PR TITLE
Add an `assert_eq` function to `std::assert`

### LINKED ISSUE #3242: Create an `assert_eq` function that checks for equality and reverts with the given value if inequality is found
I'd really like to see an `assert_eq` that reverted with the given value if inequality is found.

So in this example:

```rust
script;

use std::assert::assert_eq;

struct Data {
	value: u64,
}

fn main() {
	let data = Data {
		value: 7,
	};
	assert_eq(data.value, 1);
}
```

This would revert with the value "7". I haven't thought too hard about what this would look like for more complicated cases like structs and such though.

### PR DESCRIPTION
**Pull request by nfurfaro**
closes #3242

### CODE REVIEW

File: sway-lib-std/src/assert.sw [hunk 1]
```diff
@@ -28,3 +31,31 @@ pub fn assert(condition: bool) {
...
+/// Asserts that the given values `v1` & `v2` will always be equal during runtime.
+///
+/// ### Arguments
+///
+/// * `v1` - The first value to compare.
+/// * `v2` - The second value to compare.
+///
+/// ### Reverts
+///
+/// Reverts when `v1` != `v1`.
```
Comment (Braqzen) [inline sway-lib-std/src/assert.sw:44]:
```suggestion
/// Reverts when `v1` != `v2`.
```

File: sway-lib-std/src/assert.sw [hunk 2]
```diff
@@ -28,3 +31,31 @@ pub fn assert(condition: bool) {
...
+///
+/// ### Examples
+///
+/// ```sway
+/// fn foo(a: u64, b: u64) {
+///     assert_eq(a, b);
+///     // if code execution continues, that means a was equal to b
```
Comment (Braqzen) [inline sway-lib-std/src/assert.sw:51]:
```suggestion
///     // if code execution continues, that means `a` is equal to `b`
```

File: test/src/e2e_vm_tests/test_programs/should_pass/stdlib/assert_eq/Forc.toml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+[project]
+authors = ["Nicholas Furfaro"]
```
Comment (Braqzen) [inline test/src/e2e_vm_tests/test_programs/should_pass/stdlib/assert_eq/Forc.toml:2]:
```suggestion
authors = ["Fuel Labs <contact@fuel.sh>"]
```

Should this be updated?

File: test/src/e2e_vm_tests/test_programs/should_pass/stdlib/assert_eq/Forc.toml [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+entry = "main.sw"
+license = "Apache-2.0"
+name = "assert_eq"
+
+[dependencies]
+std = { path = "../../../../../../../sway-lib-std" }
```
Comment (Braqzen) [inline test/src/e2e_vm_tests/test_programs/should_pass/stdlib/assert_eq/Forc.toml:8]:
```suggestion
std = { path = "../../../../../../../sway-lib-std" }

```

File: test/src/e2e_vm_tests/test_programs/should_pass/stdlib/assert_eq/test.toml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+category = "run"
+expected_result = { action = "return", value = 1 }
+validate_abi = true
```
Comment (Braqzen) [inline test/src/e2e_vm_tests/test_programs/should_pass/stdlib/assert_eq/test.toml:3]:
```suggestion
validate_abi = true

```

File: test/src/e2e_vm_tests/test_programs/should_pass/stdlib/assert_eq/src/main.sw [hunk 6]
```diff
@@ -0,0 +1,76 @@
+script;
+
+use std::assert::assert_eq;
+use std::address::Address;
+use std::contract_id::ContractId;
+use std::bytes::Bytes;
+use std::vec::Vec;
+use std::identity::Identity;
+use std::b512::B512;
```
Comment (Braqzen) [inline test/src/e2e_vm_tests/test_programs/should_pass/stdlib/assert_eq/src/main.sw:9]: Some of these are in the prelude, do they need to be imported explicitly?

File: test/src/e2e_vm_tests/test_programs/should_pass/stdlib/assert_eq/src/main.sw [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+fn main() -> bool {
+
+// test_assert_eq_u64
```
Comment (Braqzen) [inline test/src/e2e_vm_tests/test_programs/should_pass/stdlib/assert_eq/src/main.sw:13]: These comments can be indented and the double `// //` can be changed to single comments

**Review by Braqzen — COMMENTED** [review]
Love the numerical comments in the error signals, now I know what they are when they come up in the SDK tests

**Review by emilyaherbert — DISMISSED** [review]
Approved from compilers side aside from @Braqzen's proposed changes.

**Review by Braqzen — DISMISSED** [review]

**Review by Braqzen — APPROVED** [review]

**Review by emilyaherbert — APPROVED** [review]

**Pull request merged**